gpytorch/models/exact_gp.py

```diff
@@ -61,6 +61,57 @@ def set_train_data(self, inputs=None, targets=None, strict=True):
             self.train_targets = targets
         self.prediction_strategy = None
 
+    def get_fantasy_model(self, inputs, targets, **kwargs):
+        if self.prediction_strategy is None:
+            raise RuntimeError("Fantasy observations can only be added after making predictions with a model so that "
+                               "all test independent caches exist. Call the model on some data first!")
+
+        if self.train_inputs[0].dim() > 2:
+            raise RuntimeError("Adding fantasy observations to a GP that is already batch mode will not be supported "
+                               "until GPyTorch supports multiple batch dimensions.")
+
+        if self.train_targets.dim() > 1:
+            raise RuntimeError("Cannot yet add fantasy observations to multitask GPs, but this is coming soon!")
+
+        if not isinstance(inputs, list):
```

Comment:
No it won't -- this is `inputs = [inputs]` not `inputs = list(inputs)`. This line is here to deal with the case where inputs is a tensor instead of a list.

Reply:
ah right my bad